src/seed/seeds.ts

```diff
@@ -0,0 +1,38 @@
+import { config } from 'dotenv';
+import { fakerPT_BR } from '@faker-js/faker';
+import { appDataSource } from '../data-source';
+import { connectedDb } from '../setup';
+import { CreateUserInput } from '../interfaces';
+import { User } from '../entity/User';
+import { hashPassword } from '../utils';
+
+async function startConnection() {
+  config();
+  console.info('Initiating setup...');
+  await connectedDb();
+}
+
+async function generateRandomData() {
+  const birthDate = fakerPT_BR.date.birthdate();
+  const password = await hashPassword(fakerPT_BR.internet.password());
+  const [firstName, lastName] = [fakerPT_BR.person.firstName(), fakerPT_BR.person.lastName()];
+  return {
+    password: password,
+    name: firstName + ' ' + lastName,
+    email: fakerPT_BR.internet.email({ firstName: firstName, lastName: lastName }),
+    birthDate: birthDate.getDay() + '/' + birthDate.getMonth() + '/' + birthDate.getFullYear(),
+  };
+}
+
+async function seedDatabase() {
+  const userRepository = appDataSource.getRepository(User);
+  await startConnection();
+
+  for (let i = 0; i < 50; i++) {
+    const newUser: CreateUserInput = await generateRandomData();
+    await userRepository.save(newUser);
+  }
```

Comment:
O `save` aceita um array de usuários tbm, e se vc enviar o array com todos, ele faz apenas uma query para o banco, enquanto aqui ele estaria fazendo 50. Além de fazer uma só ser mais rápido, num cenário remoto ele consumiria menos recurso também.

Reply:
Não tinha pensado na possibilidade de enviar um vetor de User, mas realmente faz mais sentido e é mais eficiente. 

** Acabei usando o método do seed para gerar users no teste da query de Users, para inicializar o banco de testes. Com isso, deu para notar a diferença por meio do tempo de cada teste (que é mostrado após a a conclusão de cada 'it'). Houve uma diminuição de 120ms (de ~870 para ~750) no teste mais pesado (que exige a criação de mais usuários) ao usar um save único em vez de 50.

Reply:
No caso de usar o script nos testes, eu contra-recomandaria pacotes como o faker, porque existe uma chance de ele gerar 2 usuários com o mesmo e-mail e o teste falhar esporadicamente. Pra que isso não ocorra vc pode usar um e-mail template e concatenar com o index, por exemplo.

Reply:
Entendi, não tinha pensado nesse ponto. De resto, o script de seeds seria exatamente o mesmo. Talvez adicionar uma flag para definir como gerar o email seja uma possibilidade. Não cheguei a testar, mas fiz um rascunho de como poderia ser:

```
async function generateRandomData(isTest: boolean, index: number) {
  const birthDate = fakerPT_BR.date.birthdate();
  const password = await hashPassword(fakerPT_BR.internet.password());
  const [firstName, lastName] = [fakerPT_BR.person.firstName(), fakerPT_BR.person.lastName()];
  const email: string = isTest
    ? `user${index}@test.com`
    : fakerPT_BR.internet.email({ firstName: firstName, lastName: lastName });
  return {
    password: password,
    name: firstName + ' ' + lastName,
    email: email,
    birthDate: birthDate.getDay() + '/' + birthDate.getMonth() + '/' + birthDate.getFullYear(),
  };
}

export async function seedDatabase(maxUsers = 50, isTest = false) {
  const userRepository = appDataSource.getRepository(User);
  const users = [];

  for (let i = 0; i < maxUsers; i++) {
    const newUser: CreateUserInput = await generateRandomData(isTest, i);
    users.push(newUser);
  }
  await userRepository.save(users);
}
```

Reply:
Não vejo motivo pra ter essa flag, se o seed poderia gerar sempre o modo `user${index}@test.com`.